std/bitmanip.d

```diff
@@ -72,6 +73,18 @@ private string myToString(ulong n)
     return cast(string) s ~ (n > uint.max ? "UL" : "U");
 }
 
+private template TypeName(T)
+{
+    static if(__traits(compiles, TemplateOf!T))
+    {
+        enum TypeName = T.stringof~"!"~TemplateArgsOf!T.stringof[5..$];
+    }
+    else
+    {
+        enum TypeName = T.stringof;
+    }
+}
```

Comment:
It doesn't exist because it doesn't work in the general case.  Making it work in the general case is quite hard, I have a few idea on it.  It would be a good addition to phobos but there's quite a few things to discuss on it.  Making it work well may also require additions to the language.